crates/sui-adapter/src/adapter.rs

```diff
@@ -500,16 +509,27 @@ fn process_successful_execution<
                             // with version `v+1` along with OBJECT_DIGEST_WRAPPED. Now when the object is unwrapped,
                             // it will also have version `v+1`, leading to a violation of the invariant that any
                             // object_id and version pair must be unique. Hence for any object that's just unwrapped,
-                            // we force incrementing its version number again to make it `v+2` before writing to the store.
+                            // we force it to `v+2` to make it sure it is distinct
+                            // This incrementing will be done in `delete_object`
                             state_view.log_event(Event::delete_object(
                                 module_id.address(),
                                 module_id.name(),
                                 ctx.sender(),
                                 *obj_id,
                             ));
+                            let previous_version =
+                                match state_view.get_latest_parent_entry_ref(*obj_id) {
```

Comment:
@lxfind Let me know what you think of this. Not really sure what else to do if the object isn't in parent_sync

Reply:
Good q. We probably don't want to return `ExecutionError` here. Say it failed due to a transient db failure: we don't want to charge gas for this. Feel free to fix this in a different PR, but I think we now need to distinguish between system error and execution error in this function when return (I recall you mentioned even Move execution may return system error too, so we could fix this with that together if we want)

Reply:
I think that will be a much large fix yeah, as we don't have a mechanism for such an error right now (I think). 

I'll slap a TODO on it and open an issue